subsys/bluetooth/audio/mcs.c

```diff
@@ -808,6 +809,49 @@ static void notify(const struct bt_uuid *uuid, const void *data, uint16_t len)
 	}
 }
 
+/* Helper function to notify UTF8 string values
+ * Will truncate string to fit within notification if required.
+ * The string must be null-terminated.
+ */
+static void notify_string(const struct bt_uuid *uuid, const char *str)
+{
+	/* TODO:
+	 * This function will need to get the ATT_MTU to know what length to
+	 * truncate the string to.  But the ATT_MTU is per connection, and MCS
+	 * is not connection-aware yet.
+	 * For now: Truncate according to the default ATT_MTU, so that
+	 * notifications will go through
+	 */
+
+	int err;
+	/* TODO: Use bt_gatt_get_mtu() to find the ATT_MTU */
+	const uint16_t att_mtu = 23;
+	const uint16_t maxlen = att_mtu - 1 - 2; /* Subtract opcode and handle */
+	const uint16_t len = strlen(str);
+
+	if (len <= maxlen) {
+		err = bt_gatt_notify_uuid(NULL, uuid, mcs.attrs, (void *)str, len);
+	} else {
+		/* Truncation requires a null-terminated string */
+		char trunc_str[maxlen + 1];
+
+		strncpy(trunc_str, str, maxlen);
+		trunc_str[maxlen] = '\0';
+		utf8_trunc(trunc_str);
+
+		err = bt_gatt_notify_uuid(NULL, uuid, mcs.attrs,
+					  (void *)trunc_str, strlen(trunc_str));
+	}
+
+	if (err) {
+		if (err == -ENOTCONN) {
+			BT_DBG("Notification error: ENOTCONN (%d)", err);
+		} else {
+			BT_ERR("Notification error: %d", err);
+		}
+	}
+}
```

Comment:
Should be able to do something like this for something a bit shorter. 

```suggestion
	if (len > maxlen) {
		char trunc_str[maxlen + 1];
		
		/* Truncation requires a null-terminated string */
		utf8_lcpy(trunc_str, str, maxlen);
		
		notify(uuid, (void *)trunc_str, strlen(trunc_str));
	} else {
		notify(uuid, (void *)str, len);
	}
}
```

Reply:
Much better - thanks!
I guess you mean "maxlen + 1" also in the utf8_lcpy - the termination does not go on the air.

Reply:
Good catch :)

Reply:
`sizeof()` is better, yes.  Changed.